classes/ObjectModel.php

```diff
@@ -1149,27 +1149,39 @@ public function validateField($field, $value, $id_lang = null, $skip = [], $huma
 
         // Check field validator
         if (!in_array('validate', $skip) && !empty($data['validate'])) {
-            if (!method_exists('Validate', $data['validate'])) {
-                throw new PrestaShopException($this->trans('Validation function not found: %s.', [$data['validate']], 'Admin.Notifications.Error'));
+            $callable = $data['validate'];
+            if (!is_callable($callable)) {
+                $callable = 'Validate::'. $data['validate'];
+            }
+
+            if (!is_callable($callable)) {
```

Comment:
You replaced `method_exists($a, $b)` by `is_callable($a::$b)`
Although it should be equivalent I'm afraid of php crazy behaviors with that kind of dynamic function calls 😅 

Maybe we could restore the `method_exists()` to be 100% sure there is no risk of doing that ?

Reply:
Imho, checking that the method exists proves nothing: what we want is to be able to execute a function.

Adding unit tests will ensure the behavior is still right: don't fear PHP, fear your own codebase my friend ;)

Reply:
So you mean calling both `method_exists($a, $b)` and `is_callable($a::$b)` ?

What I'm afraid is a crazy PHP behavior where `method_exists($a, $b)` returns true (and it can be used as a function) but `is_callable($a::$b)` returns false. Some edge case with nasty magic methods.

In this PR was modifying some basic class of prestashop, I would not be this paranoid but this PR modifies something core of PrestaShop 😄extra caution is needed here.

Reply:
> What I'm afraid is a crazy PHP behavior where method_exists($a, $b) returns true (and it can be used as a function) but is_callable($a::$b) returns false.

Exactly my point, if the methods exists but is not callable, the call of `call_user_func` will fail and you won't have any Exception: I'm fixing a bug here !